src/PhpOption/Some.php

```diff
@@ -132,9 +134,9 @@ public function filterIsOneOf(...$classes)
             return None::create();
         }
 
-        if (is_array($classes[0]) || $classes[0] instanceof \Traversable) {
-            $classes = $classes[0];
-        }
+        $classes = new RecursiveIteratorIterator(
```

Comment:
it largely doesn't matter but the performance of this thing is really bad. I think we should instead just do 

`$result = array();
array_walk_recursive($original_array,function($v, $k) use (&$result){ $result[] = $v; });`

Reply:
I agree, but judging from tests, this bit of code can expect that classes are passed not only as plain arrays, but as ArrayIterator as well.
`array_walk_recursive` won't walk through that.

Reply:
i see. ok well the only thing then is you need to provide unit tests that cover the scenario you were seeing.

Reply:
There is one that uses `ArrayIterator`. I suppose the idea was to allow this to traverse over any `Traversable`.
I'll add some more.